kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java

```diff
@@ -46,12 +46,10 @@ public CommandExecutor lookupKieSession(String kieSessionId, KieContainerInstanc
                 }
             }
 
-            KieServerExtension extension = registry.getServerExtension(PrometheusKieServerExtension.EXTENSION_NAME);
+            PrometheusKieServerExtension extension = (PrometheusKieServerExtension)registry.getServerExtension(PrometheusKieServerExtension.EXTENSION_NAME);
             if (extension != null && ks != null) {
                 RuleRuntimeEventManager eventManager = (RuleRuntimeEventManager)ks;
-                PrometheusMetrics metrics = PrometheusKieServerExtension.getMetrics();
-                PrometheusMetricsDroolsListener listener = new PrometheusMetricsDroolsListener(metrics, kieSessionId, containerInstance);
-                eventManager.addEventListener(listener);
+                extension.getDroolsListeners(kieSessionId, containerInstance).forEach(l -> eventManager.addEventListener(l));
```

Comment:
@cristianonicolai @lucamolteni I have a concern that we might have been and that we continue to add listeners to KieSession that already has our Prometheus listeners. Well, truth to be told, we have not seen any weird results in Prometheus tests but I am wondering how we can ensure that we are adding listeners to internal sessions/runtime only once?

Reply:
@vblagoje could you check if the default listener isnt already added to the listeners collection?